### ISSUE TITLE
perf_client will silently ignore -f flag and not output a CSV in static concurrency mode

### ISSUE THREAD

**Issue by jobegrabber** [NVIDIA/tensorrt-inference-server#513]
Unless running in dynamic concurrency mode, perf_client will not output a summary CSV and silently ignore the `-f` flag if it's provided.
I would expect either the a CSV output file, a warning or an error to be produced instead.

The CSV writing depends on whether the [`summary` vector is non-empty](https://github.com/NVIDIA/tensorrt-inference-server/blob/f7d93fd6f352ef27f8164ac3905d00bb751a34e1/src/clients/c%2B%2B/perf_client.cc#L2099) - and this vector is only appended to in dynamic concurrency mode:
```c++
if (summary.size()) { // summary gets only filled in dynamic concurrency mode
  // ...
    if (!filename.empty()) {
      // write csv
```

The CSV is produced when inserting `summary.push_back(status_summary);` [hereafter](https://github.com/NVIDIA/tensorrt-inference-server/blob/f7d93fd6f352ef27f8164ac3905d00bb751a34e1/src/clients/c%2B%2B/perf_client.cc#L2070).

**Comment by deadeyegoodwin** [comment]
I agree that we should do something. Likely we will just generate a CSV file with a single entry for the specific concurrency mode that was run.

**Comment by GuanLuo** [comment]
Fixed by #538
